### PR TITLE
conflicts: change diff line marker to %%%%%%%

### PR DESCRIPTION
**Pull request by yuja**
I feel the original -------/+++++++ pair is slightly confusing because each half can be a separator by itself. I don't know what character other than '-'/'+' is preferred, but let's pick '%' (for "mod") per @martinvonz suggestion.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)

### CODE REVIEW

File: lib/src/conflicts.rs [hunk 1]
```diff
@@ -278,27 +278,39 @@ pub fn parse_conflict(input: &[u8], num_removes: usize, num_adds: usize) -> Opti
...
             removes.push(vec![]);
         } else if line == CONFLICT_PLUS_LINE {
             plus_seen = true;
             if body_seen {
+                diff_seen = false;
                 minus_seen = false;
                 body_seen = false;
             }
             adds.push(vec![]);
-        } else if minus_seen && plus_seen {
```
Comment (yuja) [inline lib/src/conflicts.rs:301]: Do we need to support the old -------/+++++++ marker?

The current version doesn't, and the seen flags can be cleaned up by a subsequent patch.

Comment (martinvonz) [reply lib/src/conflicts.rs:301]: I don't think we need to support the old format. It's easy enough to recover.

**Review by martinvonz — APPROVED** [review]
Thanks, I think I like the new format.

**Review by arxanas — COMMENTED** [review]
I'm in favor, for what it's worth.

**Pull request merged**